File: Cpp/set/node.cpp
#include "set.h"

set::node::node() 
{
  this->value = 0;
  this->p = NULL;
  this->l = this->r = NULL;
}

set::node::node(int value) 
{
  this->value = value;
  this->p = NULL;
  this->l = this->r = NULL;
}

set::node::node(int value, set::node *l, set::node *r) {
  this->value = value;
  this->l = l;
  this->r = r;
  this->p = NULL;
}

set::node::node(const set::node &n) 
{
  this->value = n.value;
  this->l = n.l;
  this->r = n.r;
  this->p = n.p;
}

set::node::~node() 
{
}

set::node& set::node::operator = (const node &n) 
{
  this->value = n.value;
  this->l = n.l;
  this->r = n.r;
  this->p = n.p;
  return *this;
}

File: Discrete Math/3 course/Labs/02_semestr/04_ll1_generator/src/ru/ctddev/filippov/pg/Node.java
package ru.ctddev.filippov.pg;

class Node {
    protected String name;
    protected boolean isTerminal;

    Node(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        return (getClass() == o.getClass() && name.equals(((Node) o).name));
    }

    boolean isTerminal() {
        return isTerminal;
    }
}
